[Xcode 15][caffe2] Avoid redundant redeclaration of 'constexpr' static data member

cesar-vargas88 opened the pull request:
Summary: Handling the `out-of-line definition of constexpr static data member is redundant in C++17 and is deprecated [-Werror,-Wdeprecated]` warning on Xcode 15.

Test Plan: Build

Reviewed By: n0shake

Differential Revision: D46875270

malfet reviewed (changes requested):
All PyTorch builds should be done with C++17 compliant compiler, so it feels like it's better to just remove the this block.

malfet commented on aten/src/ATen/core/dynamic_type.cpp:
```diff
@@ -374,7 +374,7 @@ DynamicTypePtr ivalue::TupleTypeFactory<c10::DynamicType>::fallback(
 }
 
-TORCH_API TupleTypePtr
+C10_UNUSED TORCH_API TupleTypePtr
 ivalue::TupleTypeFactory<TupleType>::fallback(const Type& type) {
```
This is the intention, isn't it?
```suggestion
TORCH_API TupleTypePtr
ivalue::TupleTypeFactory<TupleType>::fallback(C10_UNUSED const Type& type) {
```

Commit:
[Xcode 15][caffe2] Avoid redundant redeclaration of 'constexpr' static data member (#104049)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/104049

Handling the `out-of-line definition of constexpr static data member is redundant in C++17 and is deprecated [-Werror,-Wdeprecated]` warning on Xcode 15.

Test Plan: Build

Reviewed By: tomrozanski-fb

Differential Revision: D46875270

fbshipit-source-id: 042369bd34764c9b4b2bd9ba1f1a42c22ef933f3

cesar-vargas88 commented:
@pytorchbot rebase

cesar-vargas88 commented:
@pytorchbot rebase -b main

cesar-vargas88 commented:
@pytorchbot rebase

huydhn commented:
I think we have an interesting case when the merge process was impacted by GitHub outage yesterday.  The diff D46875270 was landed internally and the merge was triggered on OSS.  However, the merge was never finished creating a divergence between OSS and fbcode.  I will merge this manually now.

huydhn commented:
@pytorchbot merge

huydhn commented:
@pytorchbot merge
